sweep/sweeper.go

```diff
@@ -1493,7 +1500,9 @@ func (s *UtxoSweeper) markInputsSwept(tx *wire.MsgTx, isOurTx bool) {
 
 		// Remove all other inputs in this exclusive group.
 		if input.params.ExclusiveGroup != nil {
-			s.removeExclusiveGroup(*input.params.ExclusiveGroup)
+			s.removeExclusiveGroup(
+				*input.params.ExclusiveGroup, outpoint,
```

Comment:
could this be a problem here that we are not removing the output which is now swept by a third party ?

Reply:
makes sense and we make sure we remove the transaction in the function `handleInputSpent` if IsOurTx=False